test/pytest_coreneuron/test_version_macros.py

```diff
@@ -11,7 +12,10 @@ def test_version_macros():
     # Get the CMake PACKAGE_VERSION variable, which should always be in
     # major.minor.patch format. This should be baked into C++ as part of the main
     # NEURON build, not as part of nrnivmodl.
-    ver = tuple(int(x) for x in h.nrnversion(0).split(".", 2))
+    # Note the PACKAGE_VERSION now is pep 440 compliant with respect to
+    # 'git describe' if possible. This results in '9.0a-1388-ga24757450' being
+    # transformed into '9.0.0a0.post1388'. So now we get the 9.0.0 from...
+    ver = tuple(int(x) for x in re.split(r"[^\d]+", h.nrnversion(0))[:3] if x)
```

Comment:
In retrospect, PEP 440 is Python-specific, and since Python only makes up a small part of NEURON, I think we could've done without it since this only makes sense for publishing wheels to PyPI...

Reply:
I kind of agree. And now I'm having trouble with certain cmake versions that apparently can't handle the regular expression specified in nrn/cmake/CheckGitDescribeCompatibility.cmake ```function(is_pep440_canonical_version)```. This  got started because of the fairly stringent space of valid tags when approaching a major.minor.patch distribution with the previous CheckGitDescribeCompatibility.cmake.
Perhaps I should pull the plug on this strategy revert back to the previous.  I was hoping the "modernization" would make tag choices more informative.

Reply:
Do we merge this then? I've bumped into the symlink issue in some other PRs, so I'd like to move on. As far as I'm concerned, the PEP-conforming changes can remain, or we can revert it in a subsequent PR.

Reply:
I'd like to merge.